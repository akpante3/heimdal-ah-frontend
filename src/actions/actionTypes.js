import keyMirror from 'keymirror';

export const ACTIONS = keyMirror({
  OPEN_MODAL: null,
  CLOSE_MODAL: null,
  SET_CURRENT_USER: null,
  SET_CURRENT_USER_FAIL: null,
  LOGOUT_USER: null,
  RESET_PASSWORD_STATE: null,
  RESET_PASSWORD_SUCCESS: null,
  RESET_PASSWORD_FAILED: null,
  RESET_PASSWORD_ERROR: null,
  UPDATE_PASSWORD_SUCCESS: null,
  UPDATE_PASSWORD_FAILED: null,
  UPDATE_PASSWORD_ERROR: null,
});
