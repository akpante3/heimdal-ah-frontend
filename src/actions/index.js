import ResetPassword from './ResetPasswordActions';

export default ResetPassword;
