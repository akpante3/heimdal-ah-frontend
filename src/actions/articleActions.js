// import axios from 'axios';
import { ACTIONS } from './actionTypes';
import errorResponse from '../utils/errorResponse';
import axiosInstance from '../utils/axiosInstance';
import { toggleLoader } from './loaderActions';

export const reportArticle = (reportData = {}, toggleModal, flashMessage) => (dispatch) => {
  const { articleId } = reportData;
  axiosInstance
    .post(`/articles/${articleId}/reports`, reportData)
    .then(() => {
      dispatch(toggleLoader());
      flashMessage();
      setTimeout(() => {
        dispatch(() => {
          toggleModal();
          return {
            type: 'ACTIONS.TOGGLE_MODAL',
          };
        });
      }, 200);
    })
    .catch((errors = {}) => {
      console.log(errors);
      console.log(errors.response);
      const { response = {}, request } = errors;

      const data = {
        dispatch,
        request,
        response,
        errors,
      };
      dispatch(toggleLoader());
      return errorResponse(data);
    });
};
