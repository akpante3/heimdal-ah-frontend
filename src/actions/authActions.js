// import axios from 'axios';
import JWT from 'jsonwebtoken';
import { ACTIONS } from './actionTypes';
import { signupEndpoint } from '../apiEndpoints';
import errorResponse from '../utils/errorResponse';
import configureLocalStorage from '../utils/configureLocalStorage';
import axiosInstance, { setToken } from '../utils/axiosInstance';

export const setAuthUser = (token) => {
  const decodedToken = JWT.decode(token);
  const { userId, username, roleId } = decodedToken;

  configureLocalStorage.setAuthUser(token);
  // remember to set axios auth token here
  setToken(token);

  return {
    type: ACTIONS.SET_AUTH_USER,
    payload: {
      user: { userId, username, roleId },
    },
  };
};

export const signupUser = (signupData = {}, history) => (dispatch) => {
  axiosInstance
    .post(signupEndpoint, signupData)
    .then((response) => {
      const { token } = response.data;
      dispatch(setAuthUser(token));
      return history.push('/');
    })
    .catch((errors = {}) => {
      const { response = {}, request } = errors;

      const data = {
        dispatch,
        request,
        response,
        errors,
      };
      return errorResponse(data);
    });
};
