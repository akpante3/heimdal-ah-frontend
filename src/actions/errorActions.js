import { SET_ERRORS, CLEAR_ERRORS, REMOVE_AN_ERROR } from './actionTypes';

export const setErrors = (payload = {}) => ({
  type: SET_ERRORS,
  payload,
});

export const clearErrors = () => ({
  type: CLEAR_ERRORS,
});

export const removeAnError = errorName => ({
  type: REMOVE_AN_ERROR,
  errorName,
});
