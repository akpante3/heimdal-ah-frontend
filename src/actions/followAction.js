import axios from 'axios';
import { FOLLOW_USER, GET_ALL_FOLLOWERS, UNFOLLOW_USER } from './actionTypes';

// const token = window.localStorage.getItem('access-token');
// const paramsId = window.localStorage.getItem('followerId');
// const followedId = parseInt(paramsId, 10);
const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjEsInVzZXJuYW1lIjoid2FsZSIsInJvbGVJZCI6MSwiaWF0IjoxNTQ4MzAzNDM5LCJleHAiOjE1NDgzODk4Mzl9.8UP-fhrGZXRUXn-xghL_gh89SBN_bO8bif1J3NM1NB0';

export const getAllFollowers = () => (dispatch) => {
  axios({
    method: 'get',
    url: 'https://heimdal-ah-staging.herokuapp.com/api/v1/users/followers',
    headers: { 'access-token': token },
  })
    .then((followData) => {
      console.log('axios=> get menu', followData);
      dispatch({
        type: GET_ALL_FOLLOWERS,
        payload: followData.data,
      });
    }).catch(err => console.log(err));
};

export const followUser = followedId => (dispatch) => {
  axios({
    method: 'post',
    url: `https://heimdal-ah-staging.herokuapp.com/api/v1/users/${followedId}/follow`,
    headers: { 'access-token': token },
  })
    .then((followData) => {
      console.log('axios=> get menu', followData);
      dispatch({
        type: FOLLOW_USER,
        // payload: followData.data,
        payload: { message: followData.data.message, followedId },
      });
    }).catch(err => console.log(err));
};

export const unfollowUser = followedId => (dispatch) => {
  axios({
    method: 'post',
    url: `https://heimdal-ah-staging.herokuapp.com/api/v1/users/${followedId}/unfollow`,
    headers: { 'access-token': token },
  })
    .then((followData) => {
      console.log('axios=> get menu', followData);
      dispatch({
        type: UNFOLLOW_USER,
        // payload: followData.data,
        payload: { message: followData.data.message, followedId },
      });
    }).catch(err => console.log(err));
};
