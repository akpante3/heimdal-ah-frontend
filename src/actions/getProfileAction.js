import axios from 'axios';
import { GET_ALL_PROFILE } from './actionTypes';

export const getAllProfile = () => (dispatch) => {
  axios
    .get('https://heimdal-ah-staging.herokuapp.com/api/v1/users')
    .then((profileData) => {
      console.log('axios=> get menu', profileData);
      dispatch({
        type: GET_ALL_PROFILE,
        payload: profileData.data,
      });
    }).catch(err => console.log(err));
};
