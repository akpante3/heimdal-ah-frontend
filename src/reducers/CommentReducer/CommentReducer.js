import { ACTIONS } from '../../actions/actionTypes';


const initialState = {
  comments: [],
  loading: false,
  status: false,
};

export default (state = initialState, action) => {
  switch (action.type) {
    case ACTIONS.DATA_LOADING_COMMENT:
      return ({
        ...state,
        loading: true,
      });

    case ACTIONS.SUCCESS_COMMENT:
      return ({
        ...state,
        status: true,
      });

    case ACTIONS.GET_ARTICLE_COMMENT: {
      const comments = action.payload;
      return ({
        ...state,
        comments,
        loading: false,
      });
    }

    case ACTIONS.POST_ARTICLE_COMMENT: {
      // note the state is the comments existing
      // while the action.payload is the newly created comment.
      let { comments } = state;
      comments = comments.concat(action.payload);
      return ({
        ...state,
        comments: comments.reverse(),
        loading: false,
      });
    }

    case ACTIONS.ARCHIVE_COMMENT: {
      let { comments } = state;
      const { commentId } = action.payload;
      comments = comments.filter(item => item.id !== commentId);
      return ({
        ...state,
        comments,
        loading: false,
      });
    }

    case ACTIONS.GET_ERRORS:
      return ({
        ...state,
        payload: action.payload,
      });

    default:
      return state;
  }
};
