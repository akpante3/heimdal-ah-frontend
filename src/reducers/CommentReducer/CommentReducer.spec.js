/* eslint-disable no-undef */
import { ACTIONS } from '../../actions/actionTypes';
import CommentReducer from './CommentReducer';


describe('select_reducer', () => {
  let initialState;
  beforeEach(() => {
    initialState = {
      loading: false,
      status: false,
      comments: undefined,
      error: '',
    };
  });

  it('testing our initial state', () => {
    const action = { type: ACTIONS.GET_ARTICLE_COMMENT };
    expect(CommentReducer(undefined, action)).toEqual(initialState);
  });

  it('is correct', () => {
    const action = { type: ACTIONS.GET_ARTICLE_COMMENT };
    expect(CommentReducer(undefined, action)).toMatchSnapshot();
  });

  it('returns the correct state', () => {
    const action = { type: ACTIONS.GET_ARTICLE_COMMENT, payload: 1 };
    expect(CommentReducer(undefined, action)).toMatchSnapshot();
  });

  it('returns the correct state', () => {
    const action = { type: ACTIONS.GET_ARTICLE_COMMENT, comments: undefined };
    const expectedState = {
      comments: undefined, loading: false, status: false, error: '',
    };
    expect(CommentReducer(initialState, action)).toEqual(expectedState);
  });
});
