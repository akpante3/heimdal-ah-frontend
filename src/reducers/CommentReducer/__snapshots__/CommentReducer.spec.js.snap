// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select_reducer is correct 1`] = `
Object {
  "comments": undefined,
  "error": "",
  "loading": false,
  "status": false,
}
`;

exports[`select_reducer returns the correct state 1`] = `
Object {
  "comments": 1,
  "error": "",
  "loading": false,
  "status": false,
}
`;
